### PR TITLE
Add a skill gain scaling setting

### LINKED ISSUE #13699: Suggestion: Configurable Learning Speed ($15 bounty) [$15]
I'd like to suggest a setting to adjust the learning speed of skills. Either in the character, world or general settings.  

Something as simple as 0.5 means all skills improve half as fast as normal, 2.0 twice as fast and so on.  Though an individual setting for each skill would be even better.

The relevant code seems to be in the SkillLevel::train function in skill.cpp.

<bountysource-plugin>
---
There is a **[$15 open bounty](https://www.bountysource.com/issues/27102165-suggestion-configurable-learning-speed-15-bounty?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Coolthulhu**
Closes #13699

Available under options->debug->right above skill rust settings. In debug, because skill rust is there. Makes sense to keep them together (both could be moved to a new "difficulty" tab, or "debug" could be renamed to that).

Scaling is a multiplier on gained xp. If scaling is not an integer, the xp gain is probabilistic.
Setting to 0.0 is a special case. Most of the changes here are around that special case:
- No xp gain means training shouldn't lower focus (because no training happens)
- Books that would grant skills shouldn't say they give skills when they couldn't due to xp settings
- Reading a skill book shouldn't lower focus equilibrium if we are ignoring that "skill" part

Training from NPCs still works as it did, because it doesn't depend on xp gain.

### CODE REVIEW

File: src/skill.cpp [hunk 1]
```diff
@@ -221,6 +228,11 @@ void SkillLevel::readBook(int minimumGain, int maximumGain, int maximumLevel)
     practice();
 }
 
+bool SkillLevel::can_train() const
```
Comment (BevapDin) [inline src/skill.cpp]: Looks like this should be a static function, or maybe even in the `Skill` class. Or do you plan to extend it using non-static members of  `SkillLevel`?

Comment (Coolthulhu) [inline src/skill.cpp]: Left it as non-static because it could be used to implement skill cap and per-skill scaling later on.

**Comment by BevapDin** [comment]
With disabled skill gain, the item description of books still says "Can bring your X skill to Y." - not sure whether this is good.

Strange thing in the character info screen: initially the skills are all white, but when moving the cursor to the skill list, they all become blue again. I think the function has actually two blocks for each of the displayed character data: one is done initially (which has been updated), the other is done when that part needs to be updated.

Looks fine otherwise.

**Comment by Marrim** [comment]
Seriously can't wait for this to go in.

**Comment by Coolthulhu** [comment]
Fixed books having "can bring your skill to X" in book description and skill drawing in `@` menu.

It is possible to disable/enable skill training at 0.0 skill rate, but it does nothing and so is not drawn.

**Comment by Zireael07** [comment]
Great!

**Comment by Midaychi** [comment]
I guess it's a little late now, but any chance of getting xp from books and xp from grinding the skill separate? That way you can disable xp gain from books but still grind the skill up manually.

**Comment by Coolthulhu** [comment]
> I guess it's a little late now, but any chance of getting xp from books and xp from grinding the skill separate?

I probably won't be updating it until I figure out a good way to set per-skill xp gain.
Removing skill gain from books could be done with a mod, though.

**Comment by kevingranade** [comment]
That's also getting into 'way too many options' territory.  If too many
options get added, I *will* periodically go through and kick them out to
mods or remove them, the number of options needs to stay manageable.

**Pull request merged**